diff --git a/dist/rules/require-test-case-name.js b/dist/rules/require-test-case-name.js
index b60e75d4ac94a725e09ba1fdd86dbd71228762b4..8bf4fbc0009eaecc429a0f54f22e454f825c66d0 100644
--- a/dist/rules/require-test-case-name.js
+++ b/dist/rules/require-test-case-name.js
@@ -54,6 +54,13 @@ const rule = {
 					hasName: false,
 					hasConfig: false
 				};
+				// We use tagged template literals to strip indentation in test cases.
+				if (testCase.type === "TaggedTemplateExpression" && testCase.quasi?.type === "TemplateLiteral") return {
+					node: testCase,
+					isObject: false,
+					hasName: false,
+					hasConfig: false
+				};
 				if (testCase.type === "ObjectExpression") {
 					let hasName = false;
 					let hasConfig = false;
diff --git a/dist/rules/test-case-shorthand-strings.js b/dist/rules/test-case-shorthand-strings.js
index 8474a7400eddb2fba2e46b71b815f237e2ee1242..c7ddefb9b16007166086246962ac44451f231565 100644
--- a/dist/rules/test-case-shorthand-strings.js
+++ b/dist/rules/test-case-shorthand-strings.js
@@ -37,6 +37,12 @@ const rule = {
 					shorthand: true,
 					needsLongform: false
 				};
+				// We use tagged template literals to strip indentation in test cases.
+				if (testCase.type === "TaggedTemplateExpression" && testCase.quasi?.type === "TemplateLiteral") return {
+					node: testCase,
+					shorthand: true,
+					needsLongform: false
+				};
 				if (testCase.type === "ObjectExpression") return {
 					node: testCase,
 					shorthand: false,
diff --git a/dist/utils.js b/dist/utils.js
index 8ab5f760911fb4441ed9c54272504bb848a1e705..04eba7ce82ef00e9b9d4188946065a6d4d8d1718 100644
--- a/dist/utils.js
+++ b/dist/utils.js
@@ -22,6 +22,8 @@ function isNormalFunctionExpression(node) {
 * @returns `true` if the node is probably constructing a RuleTester instance
 */
 function isRuleTesterConstruction(node) {
+	// We import "ruleTester" from a custom util.
+	if (node.type === "CallExpression" && node.callee.type === "Identifier" && node.callee.name === "ruleTester") return true;
 	return node.type === "NewExpression" && (node.callee.type === "Identifier" && node.callee.name === "RuleTester" || node.callee.type === "MemberExpression" && node.callee.property.type === "Identifier" && node.callee.property.name === "RuleTester");
 }
 const interestingRuleKeys = ["create", "meta"];
